mysql/awsmysql/awsmysql.go

```diff
@@ -78,27 +77,21 @@ func (uo *URLOpener) OpenMySQLURL(ctx context.Context, u *url.URL) (*sql.DB, err
 		return nil, fmt.Errorf("open RDS: empty endpoint")
 	}
 	password, _ := u.User.Password()
+	dsn := fmt.Sprintf("%s:%s@tcp(%s)%s?%s", u.User.Username(), password, u.Host, u.Path, u.RawQuery)
```

Comment:
Rather than using `fmt.Sprintf`, use `*url.URL` to construct the DSN to ensure the proper escaping.

Reply:
MySQL allows to use special characters like a `*` in password or username.
And package `url` encodes such as special characters.
So, I think we cannot use package `url` in this usecase.

Reply:
Right, but otherwise a `:` in a password will be interpreted incorrectly. The escaping issue isn't being handled here, so this seems like a step backward.